### ISSUE TITLE
Korean input bug. First character broken.

### ISSUE THREAD

**Issue by heecheolman** [ag-grid/ag-grid#3487]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**

cell input '가나다' -> 'ㄱㅏ나다'
I expected '가나다' but 'ㄱㅏ나다'

![image](https://user-images.githubusercontent.com/35059687/68135288-c2b68780-ff66-11e9-8d6e-6976fcdc6feb.png)

Related Document: https://developer.mozilla.org/en-US/docs/Mozilla/IME_handling_guide

*
*Current behavior**

Hello, I'm Korean. 
There is no problem when entering English characters in a cell. However, if you type Korean keyboard, the first letter is broken and you can enter it.
Perhaps related to the 'compositioned' event

**Expected behavior**

I hope the bug is fixed so that it doesn't break when typing Korean.

**Please tell us about your environment:**

* Mac OS
* Webstorm
* npm

* **ag-Grid version:** 21.2.2

* **Browser:** Chrome

* **Language:** [TypeScript 3.5.3]

**Comment by davis4acca** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by heecheolman** [comment]
Thank you

**Comment by davis4acca** [comment]
Hi,

We tried to reproduce this issue on our end but we could not. This issue will now be closed.

Please create a new issue in order for this to be investigated again.

If you do so, please include:

* A plunker
* Detailed step by step instructions on how to reproduce your issue
* Which specific characters on the keyboard you press and which
* What is your environment

Regards,
Davis

**Comment by icedjello** [comment]
Hi, 

After looking into this, we require a little more information before we can proceed further. 

Could you please tell us how you are triggering the edit? 

Are you double-clicking on the cell or single-clicking and then typing? 

Many thanks,

**Comment by heecheolman** [comment]
Occurs when the mouse is focused and typed in a cell. You have not entered the input mode by entering `Enter Key`.

Probably a problem with startValue in the `cellComp.ts` file. This is because English is a loose letter, and Korean and Chinese are combinations.

**Comment by icedjello** [comment]
Hi,
This is in our backlog now and we are tracking it with the following reference and description.

AG-3255 - Cell Editing + IME Edit (Korean/ Japanese): Allow capturing enter key in isolation

Note that this is parked as of this time, just because of its complexity, which means that even though we are aware of this issue, we don't have any easy solution as of this time

This error does not seem to occur when the 'enter' key is pressed prior to input. So we can suggest making sure users press 'enter' first or suppressing the keyboard event and putting the enter in yourself. 

Here is a link to suppressing keyboard events which may be useful: https://www.ag-grid.com/javascript-grid-keyboard-navigation/

Follow the status of feature requests, bugs and releases using our pipeline:

https://www.ag-grid.com/ag-grid-pipeline/
(Note that newly raised issues can take up to 2 hours to appear in the pipeline)

Please feel free to follow up if you want more updates/clarification or if you would like to follow up to check if there is aa workaround available.

Please also let us know if you don't think this feature request should be parked and your reasoning.

For more information on when can you expect your ticket to be fixed please check:

https://ag-grid.zendesk.com/hc/en-us/articles/360002213531-Where-Is-My-Ticket-

Regards

**Comment by hskim1750** [comment]
After onCellEditingStarted in cell, there is no problem when entering.
This is a problem that occurs when inputting Korean characters while being cell foscused.

              +-------------+
cell=>     | aㅏㄴ          |
              +-------------+

In the "hansontable" library, when Korean is entered in a cell, it is entered nicely.
I'll give you a reference.

Thank you.

**Comment by andyyxw** [comment]
+1

**Comment by khayashida919** [comment]
+1

**Comment by moonjoungyoung** [comment]
Hello. Any updates?

**Comment by attila-lin** [comment]
+1

**Comment by gnoeykeat** [comment]
Hello, any updates on this issue?

Based on my observations and testing, I believe this behavior may not be specific to ag-Grid, but rather related to how React handles IME-based input methods (including Korean and other IME inputs).

When processing IME inputs, React's onChange and onKeyDown events often exhibit unexpected behavior due to the asynchronous nature of IME composition events. For instance, during the IME composition process, onKeyDown may prematurely treat the key press as an English character before the IME input is finalized. As a result, during onChange, the first character might incorrectly appear as an English letter instead of the intended IME-composed character.

This issue appears to be caused by React's reconciliation and event handling model, where updates triggered by IME composition can conflict with intermediate DOM updates or default browser behaviors.

To further illustrate this behavior, I’ve prepared a simple example that demonstrates a similar issue.


https://codesandbox.io/p/sandbox/upbeat-rui-w2qs6n

![Image](https://github.com/user-attachments/assets/af6fa782-972e-4aa4-9ffe-61a1485a5381)

**Comment by khayashida919** [comment]
Please reopen this issue.

**Comment by AG-Zoheil** [comment]
Hi all,

Please note that this requirement has been added to our backlog and you can track it with the following reference: AG-3255. There is no need for the issue on GitHub to be re-opened.

We have no timeline for this at the moment, but this can still be picked up by our team when they're making modifications to this functional area of AG Grid. Please use the functionality currently provided by AG Grid until this is delivered at a later point.

You can follow the status of this item on the product pipeline page::
https://ag-grid.com/pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by zeroty** [comment]
It seems like there will continue to be issues when using composite characters. Is there any plan to fix this?

Libraries like Handsontable and ParamQuery don’t seem to have the same issues as ag-Grid.
